storage/storage.go

```diff
@@ -1057,6 +1058,37 @@ func (o *ObjectHandle) OverrideUnlockedRetention(override bool) *ObjectHandle {
 	return &o2
 }
 
+// SoftDeleted returns an object handle that can be used to get an object that
+// has been soft deleted. To get a soft deleted object, the generation must be
+// set on the object using ObjectHandle.Generation.
+// Note that an error will be returned if a live object is queried using this.
+func (o *ObjectHandle) SoftDeleted() *ObjectHandle {
+	o2 := *o
+	o2.softDeleted = true
+	return &o2
+}
+
+// Restore will restore a soft-deleted object to full object status.
+// Note that you must specify a generation to use this method.
+// copySourceACL indicates whether the restored object should copy the
+// access controls of the source object.
```

Comment:
Is this only relevant for buckets with fine-grained access? If UBLA is enabled, is this ignored or does it cause an error?